### PR TITLE
yoink: init at 0.5.0

### PR DESCRIPTION
**Pull request by ngysling**
## Description of changes

https://github.com/MrMarble/yoink
yoink is an app designed to help you download torrents marked as free leech in order to mantain your ratio in private trackers.

## Things done

- Built on platform(s)
  - [✅] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [✅] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [✅] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/p2p/yoink/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "MrMarble";
+    repo = "yoink";
+    rev = "v${version}";
+    hash = "sha256-9ftlAECywF4khH279h2qcSvKRDQX2I7GDQ7EYcEybL0=";
+  };
+
+  vendorSha256 = "sha256-cnfh2D/k4JP9BNlI+6FVLBFyk5XMIYG/DotUdAZaY0Q=";
```
Comment (eclairevoyant) [inline pkgs/applications/networking/p2p/yoink/default.nix:21]:
```suggestion
  vendorHash = "sha256-cnfh2D/k4JP9BNlI+6FVLBFyk5XMIYG/DotUdAZaY0Q=";  
```

File: pkgs/applications/networking/p2p/yoink/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "MrMarble";
+    repo = "yoink";
+    rev = "v${version}";
+    hash = "sha256-9ftlAECywF4khH279h2qcSvKRDQX2I7GDQ7EYcEybL0=";
+  };
+
+  vendorSha256 = "sha256-cnfh2D/k4JP9BNlI+6FVLBFyk5XMIYG/DotUdAZaY0Q=";  
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/applications/networking/p2p/yoink/default.nix:22]:
```suggestion
  meta = {
```

tacaking issue: #292468

**Review by eclairevoyant — COMMENTED** [review]
Thanks for the contribution!

Few points of feedback:
* Please put the file in `pkgs/by-name`
* Please ensure to follow our commit guidelines in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md regarding adding yourself as a maintainer and the format for commit messages

**Comment by hogcycle** [comment]
Thanks for your feedback. I'll be sure to follow the commit guidelines closer going forward.

**Comment by hogcycle** [comment]
ok, that's that. sorry for the redundant and sloppy commits. still really unfamiliar with git and couldn't get squashing to work. it should be ready to go, of course after review.

File: pkgs/by-name/yo/yoink/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "MrMarble";
+    repo = "yoink";
+    rev = "v${version}";
+    hash = "sha256-9ftlAECywF4khH279h2qcSvKRDQX2I7GDQ7EYcEybL0=";
+  };
+
+  vendorSha256 = "sha256-cnfh2D/k4JP9BNlI+6FVLBFyk5XMIYG/DotUdAZaY0Q=";
```
Comment (RaghavSood) [inline pkgs/by-name/yo/yoink/package.nix:21]:
```suggestion
  vendorHash = "sha256-cnfh2D/k4JP9BNlI+6FVLBFyk5XMIYG/DotUdAZaY0Q=";
```

File: pkgs/by-name/yo/yoink/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/MrMarble/yoink";
+    description = "Automatically download freeleech torrents";
+    license = licenses.mit;
+    maintainers = with maintainers; [ hogcycle ];
+  };
```
Comment (RaghavSood) [inline pkgs/by-name/yo/yoink/package.nix:27]:
```suggestion
  meta = {
    homepage = "https://github.com/MrMarble/yoink";
    description = "Automatically download freeleech torrents";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ hogcycle ];
  };
```

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -22379,6 +22379,12 @@
     githubId = 40352765;
     name = "Yoctocell";
   };
+  yoink = {
```
Comment (RaghavSood) [inline maintainers/maintainer-list.nix:22382]:
```suggestion
  hogcycle = {
```

**Review by RaghavSood — DISMISSED** [review]

**Comment by RaghavSood** [comment]
To fix your commit history, look into `git rebase -i` - then reorder them so that the add maintainer commit is first (just move the lines around in the resulting editor window), and squash them so you have two commits:

```
maintainers: add hogcycle
yoink: init at 0.5.0
```

**Comment by hogcycle** [comment]
Hi RaghavSood. The changes you suggested are done in previous commits, and I was advised to use licenses.mit and maintainers over lib.licenses.mit and lib.maintainers here. Is this still correct? I'll look into squashing them when I'm home.

https://github.com/NixOS/nixpkgs/issues/292468

**Comment by eclairevoyant** [comment]
`git rebase -i` should give open an editor window:

* delete the lines you don't want to include
* keep the lines you do want to include
* reorder the lines to reorder the commits. the first commit is at the top.
* replace "pick" with "s" or "squash" to squash the commit into the previous line
* then save and exit the editor

A more detailed guide is available at https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by hogcycle** [comment]
Thanks. Trust me, I have like 20 tabs open with git guides, but the history is all twisted up. I'd nuke and pave if that was an option.

**Comment by eclairevoyant** [comment]
I just use GH notifications, so I'm not getting any emails.

Anyway, if you're getting a lot of commits listed in the interactive rebase, what you can do here is go through the history using the website and pick out the commits that are relevant, ignoring everything else.

For example, for the maintainers commit, https://github.com/NixOS/nixpkgs/pull/312119/commits/cbc6ca48a9edff7fdb788b14d2fca7dd3cdfd4e4 is the original commit, and the follow up commits are https://github.com/NixOS/nixpkgs/pull/312119/commits/adac37fed57ce6c01b2b9ecbd3bb17a27efd4024 and https://github.com/NixOS/nixpkgs/pull/312119/commits/9f579bf1976d123f317ffb3ff1d207284d8fb13b.

As for the package itself, the proper init was in https://github.com/NixOS/nixpkgs/pull/312119/commits/c1ac53356fc40d84035739a2b12f0a1779baf7ce, and the rename was in https://github.com/NixOS/nixpkgs/pull/312119/commits/1ab5ef0f05c4cd807c49e8cb0d2806f941355610

Hence you could just write

```
# maintainer commits
pick cbc6ca48a9edff7fdb788b14d2fca7dd3cdfd4e4
squash adac37fed57ce6c01b2b9ecbd3bb17a27efd4024
squash 9f579bf1976d123f317ffb3ff1d207284d8fb13b

# init commits
reword c1ac53356fc40d84035739a2b12f0a1779baf7ce
squash 1ab5ef0f05c4cd807c49e8cb0d2806f941355610
```

Once you do so, you can then fix the second commit's message (which is what `reword` will initiate for you) to `yoink: init at 0.5.0`

I'll mark it as draft for now, just to avoid any pings of other users, in case the rebase goes sideways.

**Comment by hogcycle** [comment]
Ok, thanks. I'm pretty sure I was rebasing the rebase commit. It's a big mess lol. Thank you for guiding me through.

**Comment by hogcycle** [comment]
Thank you so much. I was under the impression that every commit needed to be squashed, that seems to be where my troubles came from.

**Comment by hogcycle** [comment]
Good to merge?

**Pull request merged**